core/ledger/ledger.go

```diff
@@ -159,7 +159,25 @@ func (ledger *Ledger) CommitTxBatch(id interface{}, transactions []*protos.Trans
 	writeBatch := gorocksdb.NewWriteBatch()
 	defer writeBatch.Destroy()
 	block := protos.NewBlock(transactions, metadata)
-	block.NonHashData = &protos.NonHashData{}
+
+	ccEvents := []*protos.ChaincodeEvent{}
+
+	if transactionResults != nil {
+		ccEvents = make([]*protos.ChaincodeEvent, len(transactionResults))
+		for i := 0; i < len(transactionResults); i++ {
+			if transactionResults[i].ChaincodeEvent != nil {
+				ccEvents[i] = transactionResults[i].ChaincodeEvent
+			} else {
+				//protobuf does not like elements of a repeated array to be nil
+				//we should discard empty events without chaincodeID when sending
+				//out events
+				ccEvents[i] = &protos.ChaincodeEvent{}
```

Comment:
If I am right you don't want to create a ccEvent for TXs which have no ChaincodeEvent in their TXresult. Consider the following: 
- ccEvents = make([]*protos.ChaincodeEvent, 0, len(transactionResults)) <-- init an array with zero length but enough capacity for the case when there is a ccEvent for each TX/TXresult.
- Then use 'if txRes[i].CCEvent != nil { ccEvents = append(ccEvents, txRes[i].CCEvent) }' to add an element if needed.
- This results in no nil / empty CCEvents and it is 'go-ish'.